AutomatedTesting/Gem/PythonTests/terrain/EditorScripts/TerrainHeightGradientList_AddRemoveGradientWorks.py

```diff
@@ -0,0 +1,145 @@
+        "Failed to retrieve height when two gradients exist."
+    )
+    triple_gradient_height_correct = (
+        "Successfully retrieved height when three gradients exist.",
+        "Failed to retrieve height when three gradients exist."
+    )
+    terrain_data_changed_call_count_correct = (
+        "OnTerrainDataChanged called expected number of times.",
+        "OnTerrainDataChanged call count incorrect."
+    )
+
+
+def TerrainHeightGradientList_AddRemoveGradientWorks():
+    """
+    Summary:
+    Test aspects of the TerrainHeightGradientList through the BehaviorContext and the Property Tree.
+
+    :return: None
+    """
+
+    import os
+    import math as sys_math
+
+    import azlmbr.legacy.general as general
+    import azlmbr.bus as bus
+    import azlmbr.math as math
+    import azlmbr.terrain as terrain
+    import azlmbr.editor as editor
+    import azlmbr.vegetation as vegetation
+    import azlmbr.entity as EntityId
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+    import editor_python_test_tools.pyside_utils as pyside_utils
+
+    terrain_changed_call_count = 0
+    expected_terrain_changed_calls = 0
+
+    def create_entity_at(entity_name, components_to_add, x, y, z):
+        entity_position = math.Vector3(x, y, z)
+        entity_id = editor.ToolsApplicationRequestBus(
+            bus.Broadcast, "CreateNewEntityAtPosition", entity_position, EntityId.EntityId()
+        )
+        entity = hydra.Entity(entity_name, entity_id)
+        if entity_id.IsValid():
+            print(f"{entity_name} entity Created")
+        entity.components = []
+        for component in components_to_add:
+            entity.components.append(hydra.add_component(component, entity_id))
+        return entity
+    
+    def on_terrain_changed(args):
+        nonlocal terrain_changed_call_count
+
+        terrain_changed_call_count += 1
+
+    def set_gradients_check_height(main_entity, gradient_list, expected_height, test_results):
+        nonlocal expected_terrain_changed_calls
+
+        test_tolerance = 0.01
+        gradient_list_path = "Configuration|Gradient Entities"
+
+        hydra.get_set_test(main_entity, 1, gradient_list_path, gradient_list)
+        expected_terrain_changed_calls += 1
+
+        # Give the TerrainSystem time to tick.
+        general.idle_wait_frames(1)
+
+        # Get the height at the origin.
+        height = terrain.TerrainDataRequestBus(bus.Broadcast, "GetHeightFromFloats", 0.0, 0.0, 0)
+
+        Report.result(test_results, sys_math.isclose(height, expected_height, abs_tol=test_tolerance))
+
+    helper.init_idle()
+
+    # Create a new level.
+    test_level_name = "temp_level"
+    helper.open_level("Terrain", "Base_WithTerrainWorldComponent")
+    helper.wait_for_condition(lambda: general.get_current_level_name() == "Base_WithTerrainWorldComponent", 2.0)
```

Comment:
This test *should* be able to open an existing level (i.e. Prefab/Base) without checking another level in and we can then add the needed level components to that level.

Reply:
Either way, line 92 is unnecessary.

Reply:
AFAIK, AutomatedTesting doesn't have prefabs enabled yet, I don't think a prefab-based level is going to work here.